marklogic-spark-connector/src/test/java/com/marklogic/spark/writer/classifier/AddClassificationToXmlTest.java

```diff
@@ -68,24 +79,47 @@ void noClassificationAddedWhenNoSemaphoreServerSpecified() {
     @Test
     @EnabledIfEnvironmentVariable(named = "SEMAPHORE_API_KEY", matches = ".*")
     void noHttpsSpecifiedShouldDefaultToHttpAndFail() {
-        final String apiKey = System.getenv("SEMAPHORE_API_KEY");
-        assertNotNull(apiKey);
-
         final DataFrameWriter<Row> dfw = readDocument("/marklogic-docs/java-client-intro.xml")
             .write().format(CONNECTOR_IDENTIFIER)
             .option(Options.CLIENT_URI, makeClientUri())
             .option(Options.WRITE_SPLITTER_XPATH, "/root/text/text()")
             .option(Options.WRITE_CLASSIFIER_HOST, "demo.data.progress.cloud")
             .option(Options.WRITE_CLASSIFIER_PORT, "443")
             .option(Options.WRITE_CLASSIFIER_ENDPOINT, "/cls/dev/cs1/")
-            .option(Options.WRITE_CLASSIFIER_APIKEY, apiKey)
+            .option(Options.WRITE_CLASSIFIER_APIKEY, API_KEY)
             .option(Options.WRITE_CLASSIFIER_TOKEN_ENDPOINT, "token/")
             .mode(SaveMode.Append);
 
         ConnectorException exception = assertThrowsConnectorException(dfw::save);
         assertTrue(exception.getMessage().contains("CloudException thrown fetching token"), "Unexpected error: " + exception.getMessage());
     }
 
+    @Test
+    void test() {
+        Dataset<Row> dataset = readDocument("/marklogic-docs/java-client-intro.xml")
+            .withColumn("textClassification", TEXT_CLASSIFIER.apply(new Column("content")));
+
+        assertEquals(1, dataset.count(), "Expecting 1 file");
+        assertEquals(9, dataset.collectAsList().get(0).size(),
+            "Expecting the 8 standard columns for representing a column, plus the 'textClassification' column.");
+
+        dataset.write().format(CONNECTOR_IDENTIFIER)
+            .option(Options.CLIENT_URI, makeClientUri())
+            .option(Options.WRITE_PERMISSIONS, DEFAULT_PERMISSIONS)
+            .option(Options.WRITE_URI_TEMPLATE, "/split-test.xml")
+            .option(Options.WRITE_CLASSIFIER_HOST, "demo.data.progress.cloud")
```

Comment:
You shouldn't need any of the classifier options here since the UDF is handling it.